src/main/java/seedu/address/model/room/Room.java

```diff
@@ -114,6 +103,35 @@ public void deleteTask(Task task) {
         }
     }
 
+    /**
+     * Replaces a task from the task list of this room with {@code editedTask}.
+     * The task must be in this room.
+     *
+     * @param target The task to replace.
+     * @param editedTask The modified task to replace the original.
+     * @throws TaskNotFoundException if task is not found in the task list of this room.
+     */
+    public void setTask(Task target, Task editedTask) {
```

Comment:
hmm for this right, why not just throw an exception in the method? why do you need to have a `try-catch` block to catch the exception and throw the same exception? eg. `public void setTask(Task target, Task editedTask) throws TaskNotFoundException {`